### PR TITLE
Hide Posts

### LINKED ISSUE #1147: [2.0] Hide Posts
Added in 0.19.4

### PR DESCRIPTION
**Pull request by Sjmarf**
Adds a "Hide" action to posts on instances running 0.19.4 or later. Uses [this MlemMiddleware PR](https://github.com/mlemgroup/MlemMiddleware/pull/33). Closes #1147.

### CODE REVIEW

File: Mlem/App/Utility/Extensions/Content Models/Post1Providing+Extensions.swift [hunk 1]
```diff
@@ -126,13 +155,25 @@ extension Post1Providing {
...
+    }
+    
     func blockAction(feedback: Set<FeedbackType>) -> ActionGroup {
         .init(
             label: "Block...",
             prompt: "Block User or Community?",
             color: Palette.main.negative,
             isDestructive: true,
-            icon: Icons.hide,
+            icon: "hand.raised",
```
Comment (EricBAndrews) [inline Mlem/App/Utility/Extensions/Content Models/Post1Providing+Extensions.swift]: *nit*

Should go in icons

File: Mlem/App/Utility/Extensions/Content Models/Post1Providing+Extensions.swift [hunk 2]
```diff
@@ -149,7 +190,7 @@ extension Post1Providing {
             color: Palette.main.negative,
             isDestructive: true,
             confirmationPrompt: showConfirmation ? "Really block this community?" : nil,
-            icon: Icons.hide,
+            icon: "xmark.circle.fill",
```
Comment (EricBAndrews) [inline Mlem/App/Utility/Extensions/Content Models/Post1Providing+Extensions.swift]: *nit*

Icons

File: Mlem/App/Views/Shared/Toast/ToastView.swift [hunk 3]
```diff
@@ -196,6 +199,7 @@ struct ToastView: View {
             .aspectRatio(contentMode: .fit)
             .fontWeight(.semibold)
             .symbolRenderingMode(.hierarchical)
+            .foregroundStyle(ToastView.dimmedSymbols.contains(systemName) ? .secondary : .primary)
```
Comment (EricBAndrews) [inline Mlem/App/Views/Shared/Toast/ToastView.swift:203]: *nit*

Palette?

**Review by EricBAndrews — APPROVED** [review]
LGTM 👍 

Couple minor nits, nothing blocking

File: Mlem/App/Views/Shared/Toast/ToastView.swift (hunk 3 above)
Comment (Sjmarf) [reply Mlem/App/Views/Shared/Toast/ToastView.swift:203]: Not this time - `.primary` and `.secondary` are choosing a variant of the color given in another `foregroundStyle` higher up the view hierarchy

**Pull request merged**